gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemTest.java

```diff
@@ -194,14 +195,22 @@ public void testGetDefaultPortIndicatesPortsAreNotUsed() throws Exception {
   @Test
   public void testImpsersonationServiceAccountUsed() throws Exception {
     Configuration config = new Configuration();
-    config.set("fs.gs.auth.access.token.provider.impl", TestingAccessTokenProvider.class.getName());
+    config.setClass(
+        "fs.gs.auth.access.token.provider.impl",
+        TestingAccessTokenProvider.class,
+        AccessTokenProvider.class);
     config.set(
         GCS_CONFIG_PREFIX + IMPERSONATION_SERVICE_ACCOUNT_SUFFIX.getKey(), "test-service-account");
 
     URI gsUri = new URI("gs://foobar/");
     GoogleHadoopFileSystem ghfs = new GoogleHadoopFileSystem();
 
-    assertThrows(GoogleJsonResponseException.class, () -> {ghfs.initialize(gsUri, config);});
+    assertThrows(
+        "401 Unauthorized",
```

Comment:
I don't think that this validates a message, it just specify acustom message if assert fails:
- https://junit.org/junit4/javadoc/latest/org/junit/Assert.html#assertThrows(java.lang.String,%20java.lang.Class,%20org.junit.function.ThrowingRunnable)